opendrift/models/openoil/openoil.py

```diff
@@ -1646,7 +1647,14 @@ def seed_elements(self, *args, **kwargs):
 
         if 'oil_type' in kwargs:
             if self.get_config('seed:oil_type') != kwargs['oil_type']:
-                self.set_config('seed:oil_type', kwargs['oil_type'])
+                if self.mode != Mode.Config:
```

Comment:
Doesn't this already work as expected? It is not allowed to change the oil-type on later seed's? I.e. you can seed multiple times, but you can only use one oil_type.

Reply:
I believe in fact this is a challenge for any property that can be provided to seed_command, but otherwise selected from default config.

Reply:
No in fact I believe this is only a challenge for `oil_type` and Leeway `object_type`, as other element properties are allowed to be different from different elements. And config value is here not overwritten by seed value, just serving as a backup value.

Reply:
I think we can fix this by changing the mode after the seed call. But we have to be careful, because we must make sure the environment is finalized. I can take a look.

Reply:
Yes, probably better than storing an internal flag `oil_type_already_chosen`
Though, repeated seed calls must be allowed, so changing mode might be problematic.

Reply:
That would allow it, but only with the same oil type.

Reply:
I added a __set_seed_config__ method that requires some special care (by us). We have to guarantee ourselves that it doesn't change the environment for instance.